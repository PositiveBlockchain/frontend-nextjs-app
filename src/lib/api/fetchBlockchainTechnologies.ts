import { fetchSheetData } from "@/lib/google";
import { config } from "@/configuration";
import { CountryData } from "@/types";

const { dataTables } = config.constants.google.sheets.databaseSheet.sheets;
const { SPREADSHEET_ID } = config.constants.google.sheets.databaseSheet;
const { name: sheetName, ranges } = dataTables;

/**
 * Fetches the possible (supported) blockchain technologies from the Google Sheet
 * @returns {Promise<string[]>} Unsorted blockchain technologies array.
 */
export async function fetchBlockchainTechnologies(): Promise<string[]> {
  const data = (await fetchSheetData({
    spreadsheetId: SPREADSHEET_ID,
    range: `${sheetName}!${ranges.blockchainTechnoloy}`,
  })) as any;

  return data.flat();
}
