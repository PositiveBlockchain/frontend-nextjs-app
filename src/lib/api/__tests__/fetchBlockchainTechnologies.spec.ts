import { fetchBlockchainTechnologies } from "../fetchBlockchainTechnologies";

describe("fetchBlockchainTechnologies", () => {
  let types: any = [];

  beforeAll(async () => {
    types = await fetchBlockchainTechnologies();
  });

  it("should match snapshot", () => {
    expect(types).toMatchSnapshot();
  });

  it("should return the region data", async () => {
    expect(types).toEqual(
      expect.arrayContaining(["Bitcoin", "Cardano", "Ethereum"])
    );
  });
});
