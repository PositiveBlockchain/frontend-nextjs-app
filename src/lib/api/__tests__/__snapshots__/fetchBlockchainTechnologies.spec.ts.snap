// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchBlockchainTechnologies should match snapshot 1`] = `
[
  "- Agnostic",
  "- Not Known",
  "- Other",
  "Algorand",
  "Aragon",
  "Atala PRISM",
  "BigchainDB",
  "Bitcoin",
  "Bitfury",
  "BNB Chain",
  "Cardano",
  "Celo",
  "Chainlink",
  "Chia",
  "Corda",
  "Cosmos",
  "DASH",
  "Energy Web Chain",
  "EOS",
  "Ethereum",
  "Gnosis chain",
  "Hedera",
  "Hyperledger",
  "IOTA",
  "IPFS",
  "Ixo Protocol",
  "KSI Blockchain",
  "NEAR Protocol",
  "NEM",
  "NEO",
  "Omisego",
  "Polkadot",
  "Polygon",
  "Regen Ledger",
  "SAP Blockchain",
  "SettleMint",
  "Sia",
  "Solana",
  "SolarCoin",
  "STEEM",
  "Stellar",
  "Tendermint",
  "Tezos",
  "VeChain",
]
`;
